### PR TITLE
RPN-933

### PR DESCRIPTION
**Pull request by mystdeim**

### CODE REVIEW

File: nft/migration/src/test/java/com/rarible/protocol/nft/migration/CreateAttributes.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import scalether.abi.Uint256Type
+import scalether.util.Hex
+import java.io.File
+import java.math.BigInteger
+
+val rest = RestTemplate()
+val filename = "data.csv"
+
+// Creates the dataset with punks attributes
+fun main() {
```
Comment (serejke) [inline nft/migration/src/test/java/com/rarible/protocol/nft/migration/CreateAttributes.kt:14]: is this file needed? maybe it's used for local experiments?

File: nft/migration/src/test/java/com/rarible/protocol/nft/migration/integration/TestPropertiesConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    private val kittiesCacheDescriptor = KittiesCacheDescriptor(86400)
+    private val yInsureCacheDescriptor = YInsureCacheDescriptor(sender, "0x181aea6936b407514ebfc0754a37704eb8d98f91", 86400, "0x1776651F58a17a50098d31ba3C3cD259C1903f7A", "http://localhost:8080")
+    private val hegicCacheDescriptor = HegicCacheDescriptor(sender, "0xcb9ebae59738d9dadc423adbde66c018777455a4", 86400, "http://localhost:8080")
+    private val hashmasksCacheDescriptor = HashmasksCacheDescriptor(sender, "0xc2c747e0f7004f9e8817db2ca4997657a7746928", 86400)
+    private val waifusionCacheDescriptor = WaifusionCacheDescriptor(sender, "0x2216d47494e516d8206b70fca8585820ed3c4946", 86400, "https://ipfs.io/ipfs/QmQuzMGqHxSXugCUvWQjDCDHWhiGB75usKbrZk6Ec6pFvw")
+
+    @Bean
+    fun itemPropService(): ItemPropertiesService {
+        return ItemPropertiesService(
+            propertiesCacheDescriptor,
```
Comment (serejke) [inline nft/migration/src/test/java/com/rarible/protocol/nft/migration/integration/TestPropertiesConfiguration.kt:50]: I guess you can `mockk()` out all these descriptors

**Review by serejke — CHANGES_REQUESTED** [review]

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/repository/item/ItemPropertyRepository.kt [hunk 3]
```diff
@@ -40,9 +40,22 @@ class ItemPropertyRepository(
             .map { it.getString(PROPERTIES_FILED) }
     }
 
+    fun get(itemId: ItemId): Mono<String> {
```
[the remark this answers is not in the document]
Comment (mystdeim) [reply nft/core/src/main/kotlin/com/rarible/protocol/nft/core/repository/item/ItemPropertyRepository.kt:43]: Just don't want to add implicit logic for duration into `get(itemId, timeToLive)`

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/repository/item/ItemPropertyRepository.kt [hunk 4]
```diff
@@ -40,9 +40,22 @@ class ItemPropertyRepository(
...
+        val query = Query().apply {
+            addCriteria(Criteria.where("_id").isEqualTo(itemId.stringValue))
+        }
+        return mongo
+            .findOne<Document>(query, COLLECTION)
+            .map { it.getString(PROPERTIES_FILED) }
+    }
+
+    fun count(): Mono<Long> {
```
[the remark this answers is not in the document]
Comment (mystdeim) [reply nft/core/src/main/kotlin/com/rarible/protocol/nft/core/repository/item/ItemPropertyRepository.kt]: Removed

File: nft/migration/src/test/java/com/rarible/protocol/nft/migration/CreateAttributes.kt (hunk 1 above)
Comment (mystdeim) [reply nft/migration/src/test/java/com/rarible/protocol/nft/migration/CreateAttributes.kt]: removed the file

File: nft/migration/src/test/java/com/rarible/protocol/nft/migration/integration/TestPropertiesConfiguration.kt (hunk 2 above)
Comment (mystdeim) [reply nft/migration/src/test/java/com/rarible/protocol/nft/migration/integration/TestPropertiesConfiguration.kt:49]: fixed

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/meta/ItemPropertiesService.kt [hunk 5]
```diff
@@ -28,6 +37,9 @@ class ItemPropertiesService(
     private val openseaClient: OpenseaClient,
     private val ipfsService: IpfsService,
     private val temporaryItemPropertiesRepository: TemporaryItemPropertiesRepository,
+    private val itemPropertiesRepository: ItemPropertyRepository,
```
[the remark this answers is not in the document]
Comment (mystdeim) [reply nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/meta/ItemPropertiesService.kt:40]: Eugene wanted to simplify a getting meta process for punks.

**Review by serejke — APPROVED** [review]

File: nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/meta/ItemPropertiesService.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (serejke) [reply nft/api/src/main/kotlin/com/rarible/protocol/nft/api/service/item/meta/ItemPropertiesService.kt:40]: I mean to keep the approach but sintatically move code to a dedicated class, to reduce clutter in the ItemPropertiesService

**Pull request merged**